### PR TITLE
Convert Link to TS

### PR DESCRIPTION
**Pull request by winkerVSbecks**

### CODE REVIEW

File: src/components/Link.stories.tsx [hunk 1]
```diff
@@ -19,7 +20,7 @@ export default {
   component: Link,
 };
 
-export const Basic = (args) => <Link {...args} />;
+export const Basic = (args: LinkProps) => <Link {...args} />;
```
Comment (ndelangen) [inline src/components/Link.stories.tsx:23]: please prefer using the FunctionComponent<Props>

```suggestion
export const Basic: FunctionComponent<LinkProps> = (props) => <Link {...props} />;
```

File: src/components/Link.tsx [hunk 2]
```diff
@@ -127,78 +135,44 @@ const LinkButton = styled.button`
...
-  ) => {
-    // Use the UnstyledLink here to avoid duplicating styles and creating
-    // specificity conflicts by first rendering the StyledLink higher up the chain
-    // and then re-rendering it through the 'as' prop.
+/**
+ * Links can contains text and/or icons. Be careful using only icons, you must provide a text alternative via aria-label for accessibility.
+ */
+export type LinkProps = React.ComponentPropsWithoutRef<'a'> &
+  React.ComponentPropsWithoutRef<'button'> &
+  StyledLinkProps & {
```
Comment (ndelangen) [inline src/components/Link.tsx:143]: You should be able to use `ComponentProps<StyledLink>`

**Review by ndelangen — CHANGES_REQUESTED** [review]

**Comment by kylesuss** [comment]
I defer to @ndelangen on this as he is much more experienced w/ TS than I.

**Comment by ndelangen** [comment]
LMK if you want to pair on this @winkerVSbecks ?

File: src/components/Link.tsx (hunk 2 above)
Comment (winkerVSbecks) [reply src/components/Link.tsx:143]: I couldn't get that working. I get an error: `'StyledLink' refers to a value, but is being used as a type here. Did you mean 'typeof StyledLink'`

**Comment by winkerVSbecks** [comment]
@ndelangen updated with your suggestions. LMK if this looks good, otherwise would love to pair.

File: src/components/Link.stories.tsx [hunk 3]
```diff
@@ -3,7 +3,8 @@ import styled from 'styled-components';
 import { action } from '@storybook/addon-actions';
 
 import { Icon } from './Icon';
-import { Link } from './Link';
+import { Link, LinkProps } from './Link';
```
Comment (ndelangen) [inline src/components/Link.stories.tsx:6]: if possible use `ComponentProps<typeof Link>` over exporting `LinkProps`

File: src/components/Link.tsx [hunk 4]
```diff
@@ -1,12 +1,12 @@
+/* eslint-disable react/prop-types */
```
Comment (ndelangen) [inline src/components/Link.tsx:1]: We can tell eslint that this rule doesn't need to be applied at all on .tsx files

File: src/components/Link.tsx [hunk 5]
```diff
@@ -21,7 +21,15 @@ const LinkInner = styled.span`
...
 `;
 
-const StyledLink = styled.a`
+type StyledLinkProps = {
+  containsIcon?: boolean;
+  secondary?: boolean;
+  tertiary?: boolean;
+  nochrome?: boolean;
+  inverse?: boolean;
+};
```
Comment (ndelangen) [inline src/components/Link.tsx:30]:
```suggestion
interface StyledLinkProps {
  containsIcon?: boolean;
  secondary?: boolean;
  tertiary?: boolean;
  nochrome?: boolean;
  inverse?: boolean;
};
```

**Review by ndelangen — APPROVED** [review]

File: src/components/Link.tsx (hunk 5 above)
Comment (ndelangen) [reply src/components/Link.tsx:30]: Looks great, some optional improvements, if agree with them 🎉  👍

**Comment by winkerVSbecks** [comment]
@ndelangen one more update. Added the `LinkComponentPicker` back in. Removing it caused unnecessary props to be padded into the `LinkWrapper`.

**Review by ndelangen — APPROVED** [review]

**Comment by ndelangen** [comment]
good to merge!

**Comment by kylesuss** [comment]
:rocket: PR was released in [`v5.2.0`](https://github.com/storybookjs/design-system/releases/tag/v5.2.0) :rocket:

**Pull request merged**